File: examples/ts/single-agent-completion.ts
import { AgenticExecutor, DisperslClient, type ToolCall } from "@codefundi/dispersl-sdk";

const client = new DisperslClient({
  baseUrl: process.env.DISPERSL_API_URL ?? "https://api.dispersl.com/v1",
  apiKey: process.env.DISPERSL_API_KEY ?? ""
});

async function toolRunner(tool: ToolCall) {
  if (tool.function.name === "end_session") {
    return { toolName: "end_session", status: "success" as const, output: "{}" };
  }
  return { toolName: tool.function.name, status: "success" as const, output: tool.function.arguments };
}

async function main() {
  const executor = new AgenticExecutor(client, toolRunner);
  const out = await executor.runAgentCompletionLoop({
    nameId: "architect",
    prompt: "Review backend resilience and produce concrete hardening actions.",
    maxLoops: 50
  });
  console.log({ taskId: out.taskId, events: out.events.length, tools: out.toolResults.length });
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: examples/py/single_agent_completion.py
import asyncio

from dispersl import AgenticExecutor, AsyncDisperslClient, ToolResult


async def tool_runner(tool: dict) -> ToolResult:
    return ToolResult(
        tool_name=tool["function"]["name"],
        status="success",
        output=tool["function"]["arguments"],
    )


async def main() -> None:
    client = AsyncDisperslClient(
        base_url="https://api.dispersl.com/v1",
        api_key="YOUR_API_KEY",
    )
    executor = AgenticExecutor(client, tool_runner)
    out = await executor.run_agent_completion_loop(
        name_id="architect",
        prompt="Review backend resilience and produce concrete hardening actions.",
        max_loops=50,
    )
    print({"task_id": out["task_id"], "events": len(out["events"]), "tools": len(out["tool_results"])})
    await client.aclose()


if __name__ == "__main__":
    asyncio.run(main())
